examples/lakers-c-native/main.c

```diff
@@ -0,0 +1,199 @@
+static const uint8_t CRED_R[] = {0xA2, 0x02, 0x60, 0x08, 0xA1, 0x01, 0xA5, 0x01, 0x02, 0x02, 0x41, 0x0A, 0x20, 0x01, 0x21, 0x58, 0x20, 0xBB, 0xC3, 0x49, 0x60, 0x52, 0x6E, 0xA4, 0xD3, 0x2E, 0x94, 0x0C, 0xAD, 0x2A, 0x23, 0x41, 0x48, 0xDD, 0xC2, 0x17, 0x91, 0xA1, 0x2A, 0xFB, 0xCB, 0xAC, 0x93, 0x62, 0x20, 0x46, 0xDD, 0x44, 0xF0, 0x22, 0x58, 0x20, 0x45, 0x19, 0xE2, 0x57, 0x23, 0x6B, 0x2A, 0x0C, 0xE2, 0x02, 0x3F, 0x09, 0x31, 0xF1, 0xF3, 0x86, 0xCA, 0x7A, 0xFD, 0xA6, 0x4F, 0xCD, 0xE0, 0x10, 0x8C, 0x22, 0x4C, 0x51, 0xEA, 0xBF, 0x60, 0x72};
+static const BytesP256ElemLen I = {0xfb, 0x13, 0xad, 0xeb, 0x65, 0x18, 0xce, 0xe5, 0xf8, 0x84, 0x17, 0x66, 0x08, 0x41, 0x14, 0x2e, 0x83, 0x0a, 0x81, 0xfe, 0x33, 0x43, 0x80, 0xa9, 0x53, 0x40, 0x6a, 0x13, 0x05, 0xe8, 0x70, 0x6b};
+
+static coap_context_t *ctx = NULL;
+static coap_session_t *session = NULL;
+static int has_coap_response = 0;
+static uint8_t coap_response_payload[MAX_MESSAGE_SIZE_LEN];
+static uint8_t coap_response_payload_len;
+
+void print_hex(uint8_t *arr, size_t len)
+{
+    printf("%ld bytes: ", len);
+    for (int i = 0; i < len; i++)
+        printf("%02X", arr[i]);
+    printf("\n");
+}
+
+static coap_response_t message_handler(coap_session_t *session COAP_UNUSED,
+                                       const coap_pdu_t *sent,
+                                       const coap_pdu_t *received,
+                                       const coap_mid_t id COAP_UNUSED)
+{
+    has_coap_response = 1;
+    // coap_show_pdu(COAP_LOG_WARN, received);
+    const uint8_t *data;
+    coap_get_data(received, &coap_response_payload_len, &data);
+    memcpy(coap_response_payload, data, coap_response_payload_len);
+    puts("received coap response");
+    print_hex((uint8_t *)coap_response_payload, coap_response_payload_len);
+    return COAP_RESPONSE_OK;
+}
+
+int coap_send_edhoc_message(uint8_t *edhoc_msg, size_t edhoc_msg_len, uint8_t value_to_prepend)
+{
+    printf("sending coap message of size %d+1\n", edhoc_msg_len);
+    coap_pdu_t *pdu = coap_pdu_init(COAP_MESSAGE_CON,
+                                    COAP_REQUEST_CODE_GET,
+                                    coap_new_message_id(session),
+                                    coap_session_max_pdu_size(session));
+    coap_add_option(pdu, COAP_OPTION_URI_PATH, 17, (const uint8_t *)".well-known/edhoc");
+    uint8_t payload[MAX_MESSAGE_SIZE_LEN];
+    payload[0] = value_to_prepend;
+    memcpy(payload + 1, edhoc_msg, edhoc_msg_len);
+    print_hex(payload, edhoc_msg_len+1);
+    coap_add_data(pdu, edhoc_msg_len + 1, payload);
+    // coap_show_pdu(COAP_LOG_WARN, pdu);
+    if (coap_send(session, pdu) == COAP_INVALID_MID)
+    {
+        coap_log_err("cannot send CoAP pdu\n");
+        return -1;
+    }
+    while (has_coap_response == 0)
+        coap_io_process(ctx, COAP_IO_WAIT);
```

Comment:
please add curly brackets even though there is a single instruction within the loop, strong opinion :)